tests/v2/validation/rbac/globalrolesv2/globalroles_v2_test.go

```diff
@@ -0,0 +1,484 @@
+
+	return grbName, clusterCount
+}
+
+func (gr *GlobalRolesV2TestSuite) TestInheritedClusterRoles() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with inheritedClusterRoles.")
+	inheritedClusterRoles := []string{roleOwner}
+	createdGlobalRole, err := createGlobalRoleWithInheritedClusterRoles(gr.client, inheritedClusterRoles)
+	require.NoError(gr.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+
+	gr.testValidateRBACResources(createdUser, inheritedClusterRoles)
+}
+
+func (gr *GlobalRolesV2TestSuite) TestMultipleInheritedClusterRoles() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with inheritedClusterRoles.")
+	inheritedClusterRoles := []string{roleCrtbView, roleProjectsCreate, roleProjectsView}
+	createdGlobalRole, err := createGlobalRoleWithInheritedClusterRoles(gr.client, inheritedClusterRoles)
+	require.NoError(gr.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+
+	gr.testValidateRBACResources(createdUser, inheritedClusterRoles)
+}
+
+func (gr *GlobalRolesV2TestSuite) TestInheritedClusterRolesWithCustomClusterRole() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a custom role that inherits rules from the cluster-owner.")
+	inheritedRoleTemplateName := namegen.AppendRandomString("crole")
+	inheritedRoleTemplate, err := gr.client.Management.RoleTemplate.Create(&management.RoleTemplate{
+		Context:         "cluster",
+		Name:            inheritedRoleTemplateName,
+		RoleTemplateIDs: []string{roleOwner},
+	})
+	require.NoError(gr.T(), err)
+
+	log.Info("Create a global role with inheritedClusterRoles.")
+	inheritedClusterRoles := []string{inheritedRoleTemplate.ID}
+	createdGlobalRole, err := createGlobalRoleWithInheritedClusterRoles(gr.client, inheritedClusterRoles)
+	require.NoError(gr.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+
+	_, expectedClusterCount := gr.testValidateRBACResources(createdUser, inheritedClusterRoles)
+
+	log.Info("Verify that the user can list all the downstream clusters.")
+	userClient, err := gr.client.AsUser(createdUser)
+	require.NoError(gr.T(), err)
+	clusterList, err := userClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
```

Comment:
is this doing something different than the `listDownstreamClusters` public function you created?

Reply:
Good catch! Modified the tests doing this to use the extension **listDownstreamClusters** instead